### PR TITLE
BC push misc fixes

### PR DESCRIPTION
**Pull request by igormukhingmailcom**

### CODE REVIEW

File: src/Client/Client.php [hunk 1]
```diff
@@ -124,8 +125,12 @@ public function pushOrder(OrderInterface $order): string
 
             $lineNumber += 10000;
 
-            // add any positive adjustments (notice we get them recursively)
-            $positiveAdjustments = $order->getAdjustmentsRecursively()->filter(static function (AdjustmentInterface $adjustment): bool {
+            // add any positive adjustments except taxes (notice we get them recursively)
+            $positiveAdjustments = $order->getAdjustmentsRecursively()->filter(static function (BaseAdjustmentInterface $adjustment): bool {
+                if (AdjustmentInterface::TAX_ADJUSTMENT === $adjustment->getType()) {
```
Comment (loevgaard) [inline src/Client/Client.php:130]: Maybe better to exclude `$adjustment->isNeutral()` adjustments?

**Review by loevgaard — COMMENTED** [review]
This is certainly a nice PR, Igor

File: src/Client/Client.php (hunk 1 above)
Comment (igormukhingmailcom) [reply src/Client/Client.php:130]: Povl asked to exclude taxes as they calculate at BC side

File: tests/Client/DTO/SalesOrderItemDataTest.php [hunk 2]
```diff
@@ -37,17 +39,31 @@ public function it_creates_from_order_item(): void
...
+        $product->addTranslation($productTranslation);
+
+        $order = new Order();
+        $order->setLocaleCode('da_DK');
 
         $orderItem = new OrderItem();
         $orderItem->setVariant($productVariant);
         $orderItem->setUnitPrice(19995);
         $orderItem->addAdjustment(self::createAdjustment(-2000));
+        $orderItem->setOrder($order);
```
Comment (loevgaard) [inline tests/Client/DTO/SalesOrderItemDataTest.php:66]: $order->addItem() instead?

File: tests/Client/DTO/SalesOrderItemDataTest.php [hunk 3]
```diff
@@ -90,9 +106,13 @@ public function setGiftCard(bool $isGiftCard): void
         $product->addVariant($productVariant);
         $product->setCode('GIFT_CARD_100');
 
+        $order = new Order();
+        $order->setLocaleCode('da_DK');
+
         $orderItem = new OrderItem();
         $orderItem->setVariant($productVariant);
         $orderItem->setUnitPrice(20000);
+        $orderItem->setOrder($order);
```
Comment (loevgaard) [inline tests/Client/DTO/SalesOrderItemDataTest.php:115]: Same here?

**Review by loevgaard — APPROVED** [review]

File: tests/Client/DTO/SalesOrderItemDataTest.php (hunk 2 above)
Comment (igormukhingmailcom) [reply tests/Client/DTO/SalesOrderItemDataTest.php:66]: It makes `$order->addItem()` inside

File: tests/Client/DTO/SalesOrderItemDataTest.php (hunk 3 above)
Comment (igormukhingmailcom) [reply tests/Client/DTO/SalesOrderItemDataTest.php:115]: See previous comment

**Pull request merged**